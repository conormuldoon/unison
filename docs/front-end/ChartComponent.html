<!DOCTYPE html>
<html lang="en">

<head>
  
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title> ChartComponent</title>

  <script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>
  <script src="https://unpkg.com/@babel/standalone/babel.min.js"></script>
  <script src="./build/entry.js"></script>
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
  <![endif]-->
  <link href="https://fonts.googleapis.com/css?family=Roboto:100,400,700|Inconsolata,700" rel="stylesheet">
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
  <link type="text/css" rel="stylesheet" href="https://jmblog.github.io/color-themes-for-google-code-prettify/themes/tomorrow-night.min.css">
  <link type="text/css" rel="stylesheet" href="styles/app.min.css">
  <link type="text/css" rel="stylesheet" href="styles/iframe.css">
  <link type="text/css" rel="stylesheet" href="">
  <script async defer src="https://buttons.github.io/buttons.js"></script>

  
</head>



<body class="layout small-header">
    <div id="stickyNavbarOverlay"></div>
    

<div class="top-nav">
    <div class="inner">
        <a id="hamburger" role="button" class="navbar-burger" aria-label="menu" aria-expanded="false">
            <span aria-hidden="true"></span>
            <span aria-hidden="true"></span>
            <span aria-hidden="true"></span>
        </a>
        <div class="logo">
            
             
                <a href="index.html">
                    <h1 class="navbar-item">Unison React components</h1>
                </a>
            
        </div>
        <div class="menu">
            
            <div class="navigation">
                <a
                    href="index.html"
                    class="link"
                >
                    API Documentation
                </a>
                
                
                
            </div>
        </div>
    </div>
</div>
    <div id="main">
        <div
            class="sidebar "
            id="sidebarNav"
        >
            
            <nav>
                
                    <h2><a href="index.html">Documentation</a></h2><div class="category"><h3>Components</h3><ul><li><a href="ARLocationComponent.html">ARLocationComponent</a></li><li><a href="ChartComponent.html">ChartComponent</a></li><li><a href="ChartPopup.html">ChartPopup</a></li><li><a href="LeafletMap.html">LeafletMap</a></li><li><a href="LocationForm.html">LocationForm</a></li><li><a href="RemoveComponent.html">RemoveComponent</a></li><li><a href="TabsComponent.html">TabsComponent</a></li><li><a href="Unison.html">Unison</a></li></ul><h3>Global</h3><ul><li><a href="global.html#DateSelector">DateSelector</a></li></ul></div>
                
            </nav>
        </div>
        <div class="core" id="main-content-wrapper">
            <div class="content">
                <header class="page-title">
                    <p>Components</p>
                    <h1>ChartComponent</h1>
                </header>
                




<section>

<header>
    
        <h2>
            &ltChartComponent <i>zoomDomain</i> <i>handleZoom</i> <i>data</i> <i>curVar</i> <i>minMax</i> <i>index</i> /&gt
        </h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        
            <div class='vertical-section'>
                <div class="members">
                    <div class="member">
                        <div class=name>
                            <span class="tag">Constructor</span>
                        </div>
                        


    
    <h4 class="name" id="ChartComponent">
        <a class="href-link" href="#ChartComponent">#</a>
        
        <span class="code-name">
            
                &ltChartComponent <i>zoomDomain</i> <i>handleZoom</i> <i>data</i> <i>curVar</i> <i>minMax</i> <i>index</i> /&gt
            
        </span>
    </h4>
    

    
    
    <div class="description">
        <p>A component for displaying line graphs.</p>
    </div>
    







    
        <h5>PropTypes:</h5>
        
<div class="table-container">
  <table class="params table">
    <thead>
    <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Required</th>
        <th>Description</th>
        <th class="last">Default</th>
    </tr>
    </thead>

    <tbody>
    
          <tr class="deep-level-0">
            <td class="name"><code>zoomDomain</code></td>
            <td>
              object
            </td>
            <td>No</td>
            <td>Specifies the beginning and end times of the time domain of the x-axis.</td>
            <td></td>
          </tr>
          
    
          <tr class="deep-level-0">
            <td class="name"><code>handleZoom</code></td>
            <td>
              func
            </td>
            <td>Yes</td>
            <td>Called when there is a zoom domain change.</td>
            <td></td>
          </tr>
          
    
          <tr class="deep-level-0">
            <td class="name"><code>data</code></td>
            <td>
              array
            </td>
            <td>Yes</td>
            <td>Data to be displayed on the graph.</td>
            <td></td>
          </tr>
          
    
          <tr class="deep-level-0">
            <td class="name"><code>curVar</code></td>
            <td>
              string
            </td>
            <td>Yes</td>
            <td>The weather variable currently selected.</td>
            <td></td>
          </tr>
          
    
          <tr class="deep-level-0">
            <td class="name"><code>minMax</code></td>
            <td>
              bool
            </td>
            <td>No</td>
            <td>True if the if displaying tabs for the precipitation variable.</td>
            <td></td>
          </tr>
          
    
          <tr class="deep-level-0">
            <td class="name"><code>index</code></td>
            <td>
              number
            </td>
            <td>No</td>
            <td>The tab index - 0 for the left tab, 1 for the middle tab, and 2 for the right tab.</td>
            <td></td>
          </tr>
          
    
    </tbody>
  </table>
</div>
    







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    
    

    

    

    
        <p class="tag-source">
            <a href="ChartComponent.js.html" class="button">View Source</a>
            <span>
                <a href="ChartComponent.js.html">ChartComponent.js</a>, <a href="ChartComponent.js.html#line21">line 21</a>
            </span>
        </p>
    
</dl>






















                    </div>
                </div>
            </div>
        
    
    </div>
    
        
    
    

    

    

    

    

    

    

    

    

    
</article>

</section>




            </div>
            
            <footer class="footer">
                <div class="content has-text-centered">
                    <p>Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.6.6</a></p>
                    <p class="sidebar-created-by">
                        <a href="https://github.com/SoftwareBrothers/better-docs" target="_blank">BetterDocs theme</a> provided with <i class="fas fa-heart"></i> by 
                        <a href="http://softwarebrothers.co" target="_blank">SoftwareBrothers - JavaScript Development Agency</a>
                    </p>
                </div>
            </footer>
            
        </div>
        <div id="side-nav" class="side-nav">
        </div>
    </div>
<script src="scripts/app.min.js"></script>
<script>PR.prettyPrint();</script>
<script src="scripts/linenumber.js"> </script>

</body>
</html>