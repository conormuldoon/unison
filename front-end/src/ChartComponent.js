import { has } from 'lodash';
import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import { VictoryAxis, VictoryChart, VictoryLine, VictoryTheme, VictoryZoomContainer } from 'victory';
import './App.css';
import { varMapping } from './Util';



const PERCENT = "Percentage";
const CELSIUS = "Celsius";

const PREVAL = "precipitation.value";

/**
 * A component for displaying line graphs.
 * 
 * @component
 * 
 */
function ChartComponent(props) {

  const [yVal, setYVal] = useState(undefined);
  const [yLabel, setYLabel] = useState(undefined);


  useEffect(() => {



    // 'Precipitation','Humidity','Wind Direction','Wind Speed','Cloudiness','Cloud Level','Dew Point','Pressure','Temperature'
    const variRequest = varMapping(props.varCur);

    if (variRequest === "precipitation") {
      if (props.minMax) {
        if (props.index === 0) {
          setYVal("precipitation.minvalue");
        } else if (props.index === 1) {
          setYVal(PREVAL);
        } else {
          setYVal("precipitation.maxvalue");
        }

      } else {
        setYVal(PREVAL);
      }
      setYLabel("Millimetres");

    } else if (variRequest === "humidity") {
      setYVal(variRequest);
      setYLabel(PERCENT);
    } else if (variRequest === "windDirection") {

      // Displaying sin of the angle on the graph, rather than the angle/
      setYVal("windDirection.sinDeg");
      setYLabel("sin of angle");
    } else if (variRequest === "windSpeed") {
      setYVal("windSpeed.mps");
      setYLabel("Metres per second");
    } else if (variRequest === "cloudiness") {
      setYVal(variRequest);
      setYLabel(PERCENT);
    } else if (variRequest === "cloudLevel") {
      if (props.index === 0) {
        setYVal("cloud.low");
      } else if (props.index === 1) {
        setYVal("cloud.medium");
      } else {
        setYVal("cloud.high");
      }
      setYLabel(PERCENT);
    } else if (variRequest === "dewPoint") {
      setYVal(variRequest);
      setYLabel(CELSIUS);
    } else if (variRequest === "pressure") {
      setYVal(variRequest);
      setYLabel("Hectopascals");
    } else if (variRequest === "temperature") {
      setYVal(variRequest);
      setYLabel(CELSIUS);
    }


  }, [props.varCur, props.minMax, props.index]);




  if (yVal && props.data && has(props.data[0], yVal)) {

    return (

      <VictoryChart style={{ parent: { maxWidth: "55%", maxHeight: "55%" } }} responsive={true} padding={{ left: 70, bottom: 50, top: 10 }}
        domainPadding={{ x: [0, 0], y: 0 }}
        theme={VictoryTheme.material}
        scale={{ x: "time" }}
        containerComponent={
          <VictoryZoomContainer
            zoomDimension="x"
            zoomDomain={props.zoomDomain}
            onZoomDomainChange={props.handleZoom}
          />
        }
      >
        <VictoryLine
          style={{
            data: { stroke: "#c43a31" },
            parent: { border: "1px solid #ccc" }
          }}
          data={props.data}
          x="date"
          y={yVal}
        />

        <VictoryAxis fixLabelOverlap={true} />
        <VictoryAxis dependentAxis label={yLabel} style={{ axisLabel: { padding: 45 } }} tickFormat={
          (x) => { if (x < .001) { return x.toFixed(3); } if (x > 999) { return Math.round(x); } return x; }} />

      </VictoryChart>

    );
  } else {

    return false;
  }

}


ChartComponent.propTypes = {
  /** Specifies the beginning and end times of the time domain of the x-axis. */
  zoomDomain: PropTypes.object,

  /** Called when there is a zoom domain change. */
  handleZoom: PropTypes.func.isRequired,

  /** Data to be displayed on the graph. */
  data: PropTypes.array,


  /** The weather variable currently selected. */
  varCur: PropTypes.string.isRequired,

  /** True if the if displaying tabs for the precipitation variable.  */
  minMax: PropTypes.bool,

  /** The tab index - 0 for the left tab, 1 for the middle tab, and 2 for the right tab. */
  index: PropTypes.number,

}

export default ChartComponent;


