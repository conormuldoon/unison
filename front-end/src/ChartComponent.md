ChartComponent example:

```js

const data=[{"date":"2019-02-23T19:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-23T20:00:00.000+0000","precipitation":{"value":0.5,"minvalue":0.0,"maxvalue":0.7}},{"date":"2019-02-23T21:00:00.000+0000","precipitation":{"value":0.6,"minvalue":0.0,"maxvalue":0.6}},{"date":"2019-02-23T22:00:00.000+0000","precipitation":{"value":3.0,"minvalue":0.0,"maxvalue":0.4}},{"date":"2019-02-23T23:00:00.000+0000","precipitation":{"value":0.2,"minvalue":0.0,"maxvalue":0.5}},{"date":"2019-02-26T18:00:00.000+0000","precipitation":{"value":0.7,"minvalue":0.0,"maxvalue":0.8}},{"date":"2019-02-25T12:00:00.000+0000","precipitation":{"value":0.3,"minvalue":0.0,"maxvalue":0.5}}];


<ChartComponent data={data} yVal={'precipitation.value'} yLabel={'Millimetres'} />


```  
