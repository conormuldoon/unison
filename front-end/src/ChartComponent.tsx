import { has } from 'lodash';
import PropTypes from 'prop-types';
import React from 'react';
import {
  VictoryAxis, VictoryChart, VictoryLine, VictoryTheme, VictoryZoomContainer,
  DomainPropType, VictoryZoomContainerProps
} from 'victory';
import './App.css';
import { varMapping } from './Util';

import { ChartData } from './ChartPopup';


const PERCENT = "Percentage";
const CELSIUS = "Celsius";

const PREVAL = "precipitation.value";


export function createChartFactory(data: ChartData, zoomDomain: DomainPropType,
  handleZoom: (domain: DomainPropType, props: VictoryZoomContainerProps) => void,
  curVar: string, minMax: boolean) {

  return function chartFactory(index: number | undefined): JSX.Element {

    return <ChartComponent data={data} zoomDomain={zoomDomain} index={index}
      handleZoom={handleZoom}
      curVar={curVar} minMax={minMax} />
  }
}

interface ChartProps {

  /**
   * The current weather variable.
   */
  curVar: string;

  /**
   * A flag that is true in cases where preciptation data is ternary.
   */
  minMax: boolean;

  /**
   * Specifies the index of the chart in cases where there are multiple chart tabs.
   */
  index: number | undefined;

  /**
   * The dat that is displayed on the chart.
   */
  data: ChartData;

  /**
   * A two element tuple that specifies the chart's zoom domain.
   */
  zoomDomain: DomainPropType;

  /**
   * A callback invoked when the zoom domain of the chart is changed.
   */
  handleZoom: (domain: DomainPropType, props: VictoryZoomContainerProps) => void;
}

/**
 * A component for displaying line graphs.
 * 
 * @component
 * 
 */
function ChartComponent({ curVar, minMax, index, data, zoomDomain, handleZoom }: ChartProps): JSX.Element | null {

  let yVal;
  let yLabel;

  // 'Precipitation','Humidity','Wind Direction','Wind Speed','Cloudiness','Cloud Level','Dew Point','Pressure','Temperature'
  const variRequest = varMapping(curVar);

  if (variRequest === "precipitation") {
    if (minMax) {
      if (index === 0) {
        yVal = "precipitation.minvalue";
      } else if (index === 1) {
        yVal = PREVAL;
      } else {
        yVal = "precipitation.maxvalue";
      }

    } else {
      yVal = PREVAL;
    }
    yLabel = "Millimetres";

  } else if (variRequest === "humidity") {
    yVal = variRequest;
    yLabel = PERCENT;
  } else if (variRequest === "windDirection") {

    // Displaying sin of the angle on the graph, rather than the angle/
    yVal = "windDirection.sinDeg";
    yLabel = "sin of angle";
  } else if (variRequest === "windSpeed") {
    yVal = "windSpeed.mps";
    yLabel = "Metres per second";
  } else if (variRequest === "cloudiness") {
    yVal = variRequest;
    yLabel = PERCENT;
  } else if (variRequest === "globalRadiation") {
    yVal = variRequest;
    yLabel = "Watts per meter squared";
  } else if (variRequest === "cloudLevel") {
    if (index === 0) {
      yVal = "cloud.low";
    } else if (index === 1) {
      yVal = "cloud.medium";
    } else {
      yVal = "cloud.high";
    }
    yLabel = PERCENT;
  } else if (variRequest === "dewPoint") {
    yVal = variRequest;
    yLabel = CELSIUS;
  } else if (variRequest === "pressure") {
    yVal = variRequest;
    yLabel = "Hectopascals";
  } else if (variRequest === "temperature") {
    yVal = variRequest;
    yLabel = CELSIUS;
  }

 
  
  if (yVal && data.length > 1 && has(data[0], yVal)) {

    return (

      <VictoryChart style={{ parent: { maxWidth: "55%", maxHeight: "55%" } }} padding={{ left: 70, bottom: 50, top: 10 }}
        domainPadding={{ x: [0, 1], y: 0 }}
        theme={VictoryTheme.material}
        scale={{ x: "time" }}
        containerComponent={
          <VictoryZoomContainer
            zoomDimension="x"
            zoomDomain={zoomDomain}
            onZoomDomainChange={handleZoom}
          />
        }
      >
        <VictoryLine
          style={{
            data: { stroke: "#c43a31" },
            parent: { border: "1px solid #ccc" }
          }}
          data={data}
          x="date"
          y={yVal}
        />

        <VictoryAxis fixLabelOverlap={true} />
        <VictoryAxis dependentAxis label={yLabel} style={{ axisLabel: { padding: 45 } }} tickFormat={
          (x) => { if (Math.abs(x) < .001) { return parseFloat(x.toFixed(3)); } if (Math.abs(x) > 999) { return Math.round(x); } return x; }} />

      </VictoryChart>

    );
  } else {

    return null;
  }

}


ChartComponent.propTypes = {
  /** Specifies the beginning and end times of the time domain of the x-axis. */
  zoomDomain: PropTypes.object,

  /** Called when there is a zoom domain change. */
  handleZoom: PropTypes.func.isRequired,

  /** Data to be displayed on the graph. */
  data: PropTypes.array.isRequired,


  /** The weather variable currently selected. */
  curVar: PropTypes.string.isRequired,

  /** True if the if displaying tabs for the precipitation variable.  */
  minMax: PropTypes.bool,

  /** The tab index - 0 for the left tab, 1 for the middle tab, and 2 for the right tab. */
  index: PropTypes.number,

}

export default ChartComponent;


