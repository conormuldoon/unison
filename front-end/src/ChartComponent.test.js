import React from 'react';
import ReactDOM from 'react-dom';
import ChartComponent from './ChartComponent';

import { render} from "react-testing-library"



const data=[{"date":"2019-02-23T19:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-23T20:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-23T21:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-23T22:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-23T23:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T18:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T12:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T13:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T14:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T15:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T16:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T17:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T18:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T19:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T20:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T21:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T22:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T23:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T00:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T01:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T02:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T03:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T04:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T05:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T06:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T07:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T08:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T09:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T10:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T11:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T12:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T13:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T14:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T15:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T16:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T17:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T00:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T01:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T02:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T03:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T04:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T05:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T06:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T07:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T08:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.1}},{"date":"2019-02-24T09:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.1}},{"date":"2019-02-24T10:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T11:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T12:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T13:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T14:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T15:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T16:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T17:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T18:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T19:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T20:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T21:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T22:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-24T23:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T00:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T01:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T02:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T03:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T04:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T05:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T06:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T07:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T08:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T09:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T10:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-25T11:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T00:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T01:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T02:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T03:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T04:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T05:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T06:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T07:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T08:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T09:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T10:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T11:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T12:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T13:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T14:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T15:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T16:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T17:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T18:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T19:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T20:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T21:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T22:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-27T23:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-28T00:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-28T01:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-28T02:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-28T03:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-28T04:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-28T05:00:00.000+0000","precipitation":{"value":0.2,"minvalue":0.1,"maxvalue":0.3}},{"date":"2019-02-26T19:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T20:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T21:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T22:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}},{"date":"2019-02-26T23:00:00.000+0000","precipitation":{"value":0.0,"minvalue":0.0,"maxvalue":0.0}}];

const chartComponent= <ChartComponent varCur='Precipitation' data={data} yVal={'precipitation.value'} yLabel={'Millimetres'} />;

it('renders without crashing', async () => {

  const div = document.createElement('div');

  ReactDOM.render(chartComponent,div);
  ReactDOM.unmountComponentAtNode(div);
});

it('mathes snapshot', () => {


  const {container} = render(chartComponent);

  expect(container).toMatchSnapshot();

});
