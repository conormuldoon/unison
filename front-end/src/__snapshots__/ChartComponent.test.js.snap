// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mathes snapshot 1`] = `<div />`;
